common/modal.js

```diff
@@ -60,6 +54,28 @@
             }
         }
 
+        // checks if error is -1, 0, 500, or 503
+        function isRetryError (error) {
+            switch (error.constructor) {
+                case ERMrest.NoConnectionError:
+                case ERMrest.TimedOutError:
+                case ERMrest.InternalServerError:
+                case ERMrest.ServiceUnavailableError:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        var vm = this;
+        vm.params = params;
+        vm.displayDetails = false;
+        vm.linkText = messageMap.showErrDetails;
+        vm.showReloadBtn = false;
+        if (ERMrest && isRetryError(exception)) {
+            vm.showReloadBtn = !$rootScope.displayReady;
```

Comment:
I think this means that we're only showing reload button on the requests that are being created before the page completely loads? Can you please add a comment as to why it's looking for `displayReady`?

Reply:
I might be wrong, but it seems like recordset is not setting this variable.

Reply:
That's exactly the point, we are only showing this for the 4 types of retriable errors while the page is loading. Last week we discussed that it doesn't make sense to "retry" other requests that may fail after the data has loaded.

I made a change to table.js to set `$rootScope.displayReady` after the main entity request succeeds or fails.

Reply:
My bad, I didn't see the `table.js` changes. Makes sense. I think we need to comment the reasoning in the code. It's not really obvious what the code is trying to do.